flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -168,7 +168,9 @@ public boolean isBounded() {
 			} catch (IOException e) {
 				throw new FlinkHiveException(e);
 			}
-			source.setParallelism(Math.min(Math.max(1, splitNum), max));
+			int parallelism = Math.min(Math.max(1, splitNum), max);
+		        parallelism = limit > 0l ? Math.min(parallelism, (int)limit) : parallelism;
```

Comment:
`(int)limit` -> maybe `limit/1000` ?
`0l` -> `0`.

Reply:
hi,jingsong:
parallelism is int type, limit is long type , If it is 0, not 0l, the system cannot be compiled,Math.min() will throw a exception. In addition, because the system default maximum parallelism  is 1000, if limit > 1000, Math.min(parallelism, (int) limit) will take the minimum of 1000, (int)limit is only to do a type cast.
I will add the comment and the Test Class later